servicetalk-loadbalancer/src/test/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancerTest.java

```diff
@@ -381,6 +400,177 @@ public void newConnectionIsClosedWhenSelectorRejects() throws Exception {
         awaitIndefinitely(connection.onClose());
     }
 
+    @Test
+    public void unhealthyHostTakenOutOfPoolForSelection() throws Exception {
+        serviceDiscoveryPublisher.onComplete();
```

Comment:
Here and in other tests, you complete `serviceDiscoveryPublisher` but later invoke `onNext`. It violates the spec. Interesting, that `TestPublisher` doesn't validate such misuse. Please, open a GH issue for that.

Reply:
Interesting, I actually copied the pattern from `selectStampedeUnsaturableConnection`, but I thought I'm re-creating the `serviceDiscoveryPublisher` in the method creating a new load balancer. But that might have gotten lost. We should improve this behaviour, you're right.

Reply:
https://github.com/apple/servicetalk/issues/1733

Reply:
Thanks for the created issue! In your current tests, please adjust the behavior to be correct (i.e. don't terminate the publisher if you intend to use it later). You can skip `onComplete()` in your tests, it's not necessary for tests that are not focused on SD termination. In general, termination of SD events stream is not expected by RRLB.

Reply:
As a matter of fact this call is necessary for tests to behave correctly whenever a new instance of the load balancer is created. The reason for it is the fact that TestPublisher prevents simultaneous subscriptions by default. Calling `onComplete()` allows the new LB to reuse the existing `serviceDiscoveryPublisher`. The `onComplete()` method is not actually shutting down the `Publisher` but its' `Subscriber`:

```
 /**
     * Completes the {@link Subscriber}.
     *
     * @see Subscriber#onComplete()
     */
    public void onComplete()
```

I think the expected changes can be discussed in the created issue.